Scripts/Runtime/Entities/TaskDriver/TaskSet/Job/JobData/CancelJobData.cs

```diff
@@ -20,13 +20,16 @@ internal CancelJobData(CancelJobConfig<TInstance> jobConfig) : base(jobConfig)
         // DATA STREAMS
         //*************************************************************************************************************
 
-        internal DataStreamCancellationUpdater<TInstance> GetDataStreamCancellationUpdater()
+        /// <summary>
+        /// Fulfills an instance of the provided type for the job.
+        /// </summary>
+        /// <param name="cancellationUpdater">The <see cref="DataStreamCancellationUpdater{TInstance}"/></param>
+        public void Fulfill(out DataStreamCancellationUpdater<TInstance> cancellationUpdater)
```

Comment:
The request side for this is implicit by the type of job.

`private struct CancelMoveJob : ITaskCancelJobForDefer`

Previously, we would feed your `Execute` function for you. 

`public void Execute(ActorMove pendingCancelInstance)`

But now because we can't use generics in the job interface type, we have to give you the index.

```csharp
public void Execute(int index)
{
                ActorMove pendingCancelInstance = m_CancellationUpdater[index];
}
```

Previously the `m_CancellationUpdater` was hidden from you. 

Now you will do:

```csharp
private DataStreamCancellationUpdater<ActorMove> m_CancellationUpdater;

            private CancelMoveJob(CancelJobData<ActorMove> jobData) : this()
            {
                jobData.Fulfill(out m_CancellationUpdater);
            }
```

Your concern of `RequestCancelFor(target)` is still the exact same because that is about requesting the start of a cancel for a certain task driver. 

This fulfill is for the job that processes elements that need custom cancel logic.

Reply:
ah ok cool. I thought this was part of the cancel request stream stuff.